gas-estimator/src/estimator.rs

```diff
@@ -0,0 +1,679 @@
+        .asset_hub_client
+        .runtime_api()
+        .at_latest()
+        .await
+        .map_err(|e| {
+            EstimatorError::InvalidCommand(format!("Failed to get latest block: {:?}", e))
+        })?
+        .call(runtime_api_call)
+        .await
+        .map_err(|e| EstimatorError::InvalidCommand(format!("Failed to dry run XCM: {:?}", e)))?;
+
+    match dry_run_result {
+        Ok(effects) => {
+            let success = matches!(effects.execution_result, Outcome::Complete { .. });
+
+            let error_message = if success {
+                None
+            } else {
+                Some(format!(
+                    "XCM execution failed: {:?}",
+                    effects.execution_result
+                ))
+            };
+
+            Ok(DryRunResult {
+                success,
+                error_message,
+                forwarded_destination: None, // TODO: Parse forwarded XCMs
+            })
+        }
+        Err(e) => Ok(DryRunResult {
+            success: false,
+            error_message: Some(format!("Dry run API error: {:?}", e)),
+            forwarded_destination: None,
+        }),
+    }
+}
+
+async fn calculate_extrinsic_fee_in_dot(
+    clients: &Clients,
+    xcm: &VersionedXcm,
+    origin: [u8; 20],
+) -> Result<u128, EstimatorError> {
+    use bridge_hub_westend_runtime::runtime_types::snowbridge_verification_primitives::EventProof;
+    let fixture = snowbridge_pallet_ethereum_client_fixtures::make_inbound_fixture();
+    let encoded_event_proof = codec::Encode::encode(&fixture.event);
+
+    let runtime_event_proof: EventProof = codec::Decode::decode(&mut &encoded_event_proof[..])
+        .map_err(|e| EstimatorError::InvalidCommand(format!("Failed to decode EventProof: {:?}", e)))?;
+
+    let submit_call = bridge_hub_westend_runtime::runtime::tx()
+        .ethereum_inbound_queue_v2()
+        .submit(runtime_event_proof);
+
+    let alice = dev::alice();
+    let unsigned_extrinsic = clients
+        .bridge_hub_client
+        .tx()
+        .create_unsigned(&submit_call)
+        .unwrap().encoded().to_vec();
+
+    //let info = clients
+    //    .bridge_hub_client
+    //    .tx()
+    //    .create_unsigned(&submit_call)
+    //    .unwrap()
+    //    .partial_fee_estimate()
+    //    .await
+    //    .unwrap();
+
+    let payment_info_call = bridge_hub_westend_runtime::runtime::apis()
+        .transaction_payment_api()
+        .query_info(unsigned_extrinsic.into(), 0); // 0 for encoded length
+    let info = clients
+        .bridge_hub_client
+        .runtime_api()
+        .at_latest()
+        .await
+        .map_err(|e| {
+            EstimatorError::InvalidCommand(format!("Failed to get latest block: {:?}", e))
+        })?
+        .call(payment_info_call)
+        .await
+        .map_err(|e| {
+            EstimatorError::InvalidCommand(format!("Failed to query delivery fees: {:?}", e))
+        })?;
+
+    Ok(info.partial_fee)
```

Comment:
The partial fee is 0, I am not sure why. Both using `.partial_fee_estimate()` and `.query_info()`

Reply:
@claravanstaden Made a fix in https://github.com/Snowfork/snowbridge/pull/1529